c-version/elrond/context.h

```diff
@@ -0,0 +1,59 @@
+#ifndef _CONTEXT_H_
+#define _CONTEXT_H_
+
+#include "types.h"
+
+void getSCAddress(byte *address);
+void getOwnerAddress(byte *address);
+int getShardOfAddress(const byte *address);
+int isSmartContract(const byte *address);
+
+// Call-related functions
+void getOriginalTxHash(byte *txHash);
+void getCaller(byte *callerAddress);
+int getFunction(byte *function);
+int getCallValue(byte *result);
+long long getGasLeft();
+void finish(const byte *data, int length);
+void int64finish(long long value);
+void writeLog(byte *pointer, int length, byte *topicPtr, int numTopics);
+void asyncCall(const byte *destination, const byte *value, const byte *data, int length);
+void signalError(byte *message, int length);
+
+int executeOnSameContext(long long gas, byte *address, byte *value, byte *function, int functionLength, int numArguments, byte *argumentsLengths, byte *arguments);
+int executeOnDestContext(long long gas, byte *address, byte *value, byte *function, int functionLength, int numArguments, byte *argumentsLengths, byte *arguments);
+int createContract(long long gas, byte *value, byte *code, int codeSize, byte *newAddress, int numInitArgs, byte *initArgLengths, byte *initArgs);
+
+// Blockchain-related functions
+long long getBlockTimestamp();
+long long getBlockNonce();
+int getBlockHash(long long nonce, byte *hash);
+
+// Argument-related functions
+int getNumArguments();
+int getArgument(int argumentIndex, byte *argument);
+long long int64getArgument(int argumentIndex);
```

Comment:
int64getArgument is deprecated
use smallIntGetUnsignedArgument instead (or smallIntGetSignedArgument, but I don't think you need signed numbers anywhere in the contract)